secp256k1-sys/build.rs

```diff
@@ -31,6 +31,38 @@ fn main() {
                // just #define it away.
                .define("printf(...)", Some(""));
 
+    if env::var("CARGO_CFG_TARGET_ARCH").unwrap() == "riscv32" {
+        const DEFAULT_RISCV_GNU_TOOLCHAIN: &str = "/opt/riscv";
+        println!("cargo:rerun-if-env-changed=RISCV_GNU_TOOLCHAIN");
+
+        let riscv_gnu_toolchain_path = env::var("RISCV_GNU_TOOLCHAIN").unwrap_or_else(|_| {
+            println!("cargo:warning=Variable RISCV_GNU_TOOLCHAIN unset. Assuming '{DEFAULT_RISCV_GNU_TOOLCHAIN}'");
+            println!("cargo:warning=Please make sure to build riscv toolchain:");
+            println!("cargo:warning=  git clone https://github.com/riscv-collab/riscv-gnu-toolchain && cd riscv-gnu-toolchain");
+            println!("cargo:warning=  export RISCV_GNU_TOOLCHAIN={DEFAULT_RISCV_GNU_TOOLCHAIN}");
+            println!("cargo:warning=  configure --prefix=\"$RISCV_GNU_TOOLCHAIN\" --with-arch=rv32im --with-abi=ilp32");
+            println!("cargo:warning=  make -j$(nproc)");
+
+            // if unset, try the default and fail eventually
+            DEFAULT_RISCV_GNU_TOOLCHAIN.into()
+        });
```

Comment:
I don't think it's appropriate to force every RISCV compilation to use the GNU toolchains - are there not other toolchains?